python/src/opendp/extras/numpy/__init__.py

```diff
@@ -117,7 +117,9 @@ def array2_domain(
     _check_nonnegative_int(size, "size")
     _check_nonnegative_int(num_columns, "num_columns")
 
-
+    if nan is None:
+        nan = T in {"f32", "f64"}
+    
     T = T or _ELEMENTARY_TYPES.get(origin.dtype.type)
```

Comment:
If the order of these lines were switched, the behavior would change:
```
T = T or _ELEMENTARY_TYPES.get(origin.dtype.type)

if nan is None:
    nan = T in {"f32", "f64"}
```
In this order, if T started as `None`, but `origin.dtype.type` was `float`, `nan` will end up `True`.

If it's correct just the way it is, maybe add a comment to explain?

Reply:
Thanks good catch.